.github/workflows/main.yml

```
@@ -28,3 +28,16 @@ jobs:
       - name: verify dafny
         run: __tests__/verify-dafny.sh ${{ matrix.dafny }}
         shell: bash
+
+  test:
```

```suggestion
  unit-tests:
```

?

Also consider renaming `setup-dafny` to something like `test-action` or `integration-tests`